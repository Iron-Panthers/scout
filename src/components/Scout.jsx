import React, { useContext, useRef } from "react"

import { Context } from "../state"
import Dropdown from "./inputs/Dropdown"
import Next from "./inputs/Next"
import Shoot from "./inputs/Shoot"
import Switch from "./inputs/Switch"
import Tabs from "./inputs/Tabs"
import Undo from "./inputs/Undo"
import Info from "./inputs/Info"
import Bool from "./inputs/Bool"
import Count from "./inputs/Count"

import "./Scout.scss"

const Scout = () => {
  const [state, dispatch] = useContext(Context)

  const phaseTabContent = (() => {
    switch (state.phase) {
      case "auto":
        return (
          <>
            <Count prop="pickup" phase label="Ball Pickup" color="green" />
            <Bool prop="taxi" phase label="Taxi" color="green" />
            <Shoot></Shoot>
          </>
        )
      case "teleop":
        return (
          <>
            <Bool
              prop="wrongCargo"
              label="Shoot Wrong Cargo"
              trueLabel="Shot Wrong Cargo"
              color="green"
            ></Bool>
            <Bool
              prop="defense"
              label="Defense"
              trueLabel="Defended"
              color="green"
            ></Bool>
            <Shoot></Shoot>
          </>
        )
      case "endgame":
        return (
          <>
            <Switch
              phase
              options={{
                opA: { label: "Climb", prop: "climb", color: "green" },
                opB: { label: "Fail", prop: "fail", color: "red" },
              }}
            ></Switch>
            <Dropdown
              wide
              phase
              prop="level"
              options={[
                "0 None",
                "1 Low",
                "2 Mid Rung",
                "3 High Rung",
                "4 Traversal Rung",
              ]}
            ></Dropdown>
          </>
        )
      default:
        return <div>no tab exists.</div>
    }
  })()

  const showNext = state.phase === "endgame"

  return (
    <>
      <div className="scoutHead">
        <Info></Info>
        <Tabs></Tabs>
      </div>
      {phaseTabContent}
      {showNext && <Next wide></Next>}
      <Undo wide></Undo>
    </>
  )
}

export default Scout
