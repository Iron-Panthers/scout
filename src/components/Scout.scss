div.scoutHead {
  grid-column: span 6;
  display: flex;
  flex-direction: column;
}
