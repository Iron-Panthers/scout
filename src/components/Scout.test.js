import Scout from "./Scout"
import React from "react"
import { screen } from "@testing-library/react"

const tabSeq = ["teleop", "endgame", "auto", "endgame", "teleop", "auto"]
// const tabs = ["auto", "teleop", "endgame"]

it("renders every tab without crashing", () => {
  renderContext(<Scout></Scout>)

  tabSeq.forEach((tab) => {
    expect(screen.getByText(tab)).toBeInTheDocument()
    screen.getByText(tab).click()
  })
})

it("disables tab when clicked", () => {
  renderContext(<Scout></Scout>)

  tabSeq.forEach((tab) => {
    expect(screen.getByText(tab)).not.toBeDisabled()
    screen.getByText(tab).click()
    expect(screen.getByText(tab)).toBeDisabled()
  })
})

it("only renders next button on endgame tab", () => {
  renderContext(<Scout></Scout>)
  ;["auto", "teleop"].forEach((tab) => {
    screen.getByText(tab).click()
    expect(screen.queryByText("NEXT")).toBeNull()
  })
  screen.getByText("endgame").click()
  expect(screen.getByText("NEXT")).toBeInTheDocument()
})

it("counts shots", () => {
  renderContext(<Scout></Scout>)

  for (let i = 0; i < 10; i++) {
    expect(screen.getByText(`Lower hub ${i}`)).toBeInTheDocument()
    screen.getByText(/Lower hub/).click()
  }
})

it("matches snapshot", () => {
  matchesSnapshot(<Scout></Scout>)
})
