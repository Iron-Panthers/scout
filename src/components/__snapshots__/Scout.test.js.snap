// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="scoutHead"
  >
    <div
      className="Info Center"
    >
      <p>
        qualification #??
      </p>
      <p>
        team #????
      </p>
    </div>
    <div
      className="Tabs wide evenFlex"
    >
      <button
        className="Tab"
        disabled={true}
        onClick={[Function]}
      >
        auto
      </button>
      <button
        className="Tab"
        disabled={false}
        onClick={[Function]}
      >
        teleop
      </button>
      <button
        className="Tab"
        disabled={false}
        onClick={[Function]}
      >
        endgame
      </button>
    </div>
  </div>,
  <button
    className="green"
    onClick={[Function]}
  >
    Ball Pickup 0
  </button>,
  <button
    className="green"
    disabled={false}
    onClick={[Function]}
  >
    Taxi
  </button>,
  <button
    className="green"
    onClick={[Function]}
  >
    Upper hub 0
  </button>,
  <button
    className="red"
    onClick={[Function]}
  >
    fail 0
  </button>,
  <button
    className="green"
    onClick={[Function]}
  >
    Lower hub 0
  </button>,
  <button
    className="red"
    onClick={[Function]}
  >
    fail 0
  </button>,
  <button
    className="wide blue"
    disabled={true}
    onClick={[Function]}
  >
    UNDO
  </button>,
]
`;
